libevmasm/EthdebugSchema.h

```diff
@@ -0,0 +1,172 @@
+/*
+	This file is part of solidity.
+
+	solidity is free software: you can redistribute it and/or modify
+	it under the terms of the GNU General Public License as published by
+	the Free Software Foundation, either version 3 of the License, or
+	(at your option) any later version.
+
+	solidity is distributed in the hope that it will be useful,
+	but WITHOUT ANY WARRANTY; without even the implied warranty of
+	MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+	GNU General Public License for more details.
+
+	You should have received a copy of the GNU General Public License
+	along with solidity.  If not, see <http://www.gnu.org/licenses/>.
+*/
+// SPDX-License-Identifier: GPL-3.0
+
+#pragma once
+
+#include <libsolutil/Common.h>
+#include <libsolutil/JSON.h>
+
+#include <concepts>
+#include <optional>
+#include <variant>
+#include <vector>
+
+namespace solidity::evmasm::ethdebug::schema
+{
+
+struct EthdebugException: virtual util::Exception {};
+
+namespace data
+{
+
+struct HexValue
+{
+	bytes value;
+};
+
+struct Unsigned
+{
+	template<std::unsigned_integral T>
+	Unsigned(T const _value)
+	{
+		solRequire(static_cast<T>(_value) <= std::numeric_limits<std::uint64_t>::max(), EthdebugException, "Too large value.");
+		value = static_cast<std::uint64_t>(_value);
+	}
+	template<std::signed_integral T>
+	Unsigned(T const _value)
+	{
+		solRequire(_value >= 0, EthdebugException, "NonNegativeValue got negative value.");
+		solRequire(static_cast<std::make_unsigned_t<T>>(_value) <= std::numeric_limits<std::uint64_t>::max(), EthdebugException, "Too large value.");
+		value = static_cast<std::uint64_t>(_value);
+	}
+	Unsigned(HexValue&& _value): value(std::move(_value)) {}
+
+	std::variant<std::uint64_t, HexValue> value;
+};
+
+}
+
+namespace materials
+{
+
+struct ID
+{
+	std::variant<std::string, std::uint64_t> value;
+};
+
+struct Reference
+{
+	enum class Type { Compilation, Source };
+	ID id;
+	std::optional<Type> type;
+};
+
+struct SourceRange
+{
+	struct Range
+	{
+		data::Unsigned length;
+		data::Unsigned offset;
+	};
+
+	Reference source;
+	std::optional<Range> range;
+};
+
+}
+
+namespace program
+{
+
+struct Context
+{
+	struct Variable
+	{
+		std::optional<std::string> identifier;
+		std::optional<materials::SourceRange> declaration;
+		// todo type
+		// todo pointer according to ethdebug/format/spec/pointer
```

Comment:
If you use lowercase it does not stand out as much as it should (and also editors don't highlight it).

```suggestion
		// TODO: type
		// TODO: pointer according to ethdebug/format/spec/pointer
```